addons/mail/static/src/utils/common/format.js

```diff
@@ -110,9 +110,13 @@ function linkify(text) {
         // Decode the url first, in case it's already an encoded url
         const url = decodeURI(match[0]);
         const href = encodeURI(!/^https?:\/\//i.test(url) ? "http://" + url : url);
-        result += `<a target="_blank" rel="noreferrer noopener" href="${href}">${_escapeEntities(
-            url
-        )}</a>`;
+        const paramsMatch = /\/mail\/([a-zA-Z.]+)\/(\d+)\/message\/(\d+\b)/.exec(url);
+        let urlParams = `target="_blank" rel="noreferrer noopener" href="${href}"`;
+        if (paramsMatch && new URL(href).origin === getOrigin()) {
+            const oeData = `data-oe-model="${paramsMatch[1]}" data-oe-res-id="${paramsMatch[2]}" data-oe-id="${paramsMatch[3]}"`;
+            urlParams += ` class='o_message_redirect' ${oeData}`;
+        }
+        result += `<a ${urlParams}>${_escapeEntities(url)}</a>`;
```

Comment:
That would work in Odoo Environment. But, outside odoo we won't be able to use it. Discord does allow to copy message links and then paste them outside, and they are properly structured there. So, IMO the links should be copied in structured format only.

Reply:
So, I suggest copy paste the structured links itself. But, the modification of these links would be done using another function instead of `linkify`. Like creating a new `transformFunction` ?

Reply:
What I was thinking, but again not an expert of Clipboard API so I'm not entirely sure this is feasible:

When using "Copy Message Link", the clipboard saves an item that contains both a special odoo structure (let's say `odoo/message_link`) and a `text/plain` version with is simply the full URL. In all other applications than Odoo, they just paste the `text/plain` version. But In Odoo, we could retrieve the `odoo/message_link` version to have richer info when posting the message. It's invisible to user when pasting before posting a message, as it dumps the URL. But there could be some JS under-the-hood like when you select a mention in suggestion where there's a map between this pasted URL which is flagged as a message link. When posting the message, `generateMentionLinks` could also prettify the message link.

Reply:
If this is feasible, I would prefer this version, because regex on URL relies too heavily on the current URL format. We changed it recently to look better, but I really don't want to shoot ourselves in the foot and have code that necessarily relies indefinitely on the current URL format. We could change our mind again in the future and invent another format for URL, and it'd be very painful to be forced to urgently adapt this feature again because we didn't design it in a way that requires less maintenance... And would just work reliably by not being coupled to the URL format at hand.

Reply:
It is possible to do what you suggested. But, to do so we would need to add an extra eventlistener on `composer` to catch `paste` events. As clipboard doesn't provide any way to know whether something got read or not. So, is it ok?

Reply:
Yes, that's be perfect with handling on paste. Thanks! ☺️

Reply:
In the end I think we should remove Clipboard stuff completely.

This was a rabbit hole from focusing too much on pretty link and the lack of thread name without fetching server sometimes. This makes the feature far more complicated than needed.

So to sum up what we really want: only show the simplest URL for any message, basically `/mail/message/58431`. Whether this comes from `discuss.channel` or chatter, no change of URL: they all look the same with just message-id. I would just open the link like any other link, in a new tab. So no jump to message in the same page. This latter aspect, I'd wait for pretty link before doing this.

And for the pretty link, it's not really that complicated: we could just transform the message during its rendering in `prepareMessageBody()` to prettify the link. We use the thread name if we have knowledge, otherwise we could display "unknown" or fetch the thread name of message, while keeping the plain and simple link `/mail/message/58431`. I would not store thread model/name/id in the URL, because it'd make the link too complex and increase maintenance cost (e.g. what if change of thread name, thread model name, or compatibility with sub-thread modelling, etc.).

I suggest to do the pretty message link in a follow-up PR, so that the core feature without pretty link is plain and simple.

Reply:
> So to sum up what we really want: only show the simplest URL for any message, basically /mail/message/58431. Whether this comes from discuss.channel or chatter, no change of URL: they all look the same with just message-id. I would just open the link like any other link, in a new tab. So no jump to message in the same page. This latter aspect, I'd wait for pretty link before doing this.

Ok.

> We use the thread name if we have knowledge, otherwise we could display "unknown" or fetch the thread name of message

The issue with this approach is that it will cause delay in the display of the message. As, we would be fetching the thread names after we get messages on client side.